Userland/Libraries/LibSoftGPU/Device.h

```diff
@@ -104,13 +119,16 @@ class Device final {
     RefPtr<Gfx::Bitmap> m_render_target;
     OwnPtr<DepthBuffer> m_depth_buffer;
     RasterizerOptions m_options;
+    LightModelParameters m_lighting_model;
     Clipper m_clipper;
     Vector<Triangle> m_triangle_list;
     Vector<Triangle> m_processed_triangles;
     Vector<Vertex> m_clipped_vertices;
     Array<Sampler, NUM_SAMPLERS> m_samplers;
     Vector<size_t> m_enabled_texture_units;
     AlphaBlendFactors m_alpha_blend_factors;
+    Vector<Light, NUM_LIGHTS> m_lights;
```

Comment:
This can be an `Array` instead, with auto-initialization as a bonus

Reply:
How I didn't make the connection this could be the case is beyond me.....

OO overload D: